src/traccuracy/metrics/_ctc.py

```diff
@@ -253,3 +255,128 @@ def get_weighted_error_sum(
         edge_error_counts, edge_fp_weight, edge_fn_weight, edge_ws_weight
     )
     return vertex_error_sum + edge_error_sum
+
+
+class CellCycleAccuracy(Metric):
+    """The CCA metric captures the ability of a method to identify a distribution of cell
+    cycle lengths that matches the distribution present in the ground truth. The evaluation
+    is done on distributions and therefore does not require a matching of solution to the
+    ground truth. It ranges from [0,1] with higher values indicating better performance.
+
+    This metric is part of the biologically inspired metrics introduced by the CTC
+    and defined in Ulman 2017.
+    """
+
+    def __init__(self) -> None:
+        valid_matching_types = ["one-to-one", "many-to-one", "one-to-many", "many-to-many"]
+        super().__init__(valid_matching_types)
+
+    def _get_lengths(self, track_graph: TrackingGraph) -> list[int]:
+        """Identifies the length of complete cell cycles in a tracking graph
+
+        Args:
+            track_graph (TrackingGraph): The graph to evaluate
+
+        Returns:
+            list[int]: a list of complete cell cycle lengths
+        """
+        lengths = []
+        # Loop over subgraphs
+        for g in track_graph.get_connected_components():
+            divs = g.get_divisions()
+            # Need at least two divisions to calculate cell cycle length
+            if len(list(divs)) >= 2:
+                lengths.extend(self._get_subgraph_lengths(g, divs))
```

Comment:
I think there should be a quicker way to do this than walking the graph. I'm thinking we first make a subgraph that filters out any connected components where there are fewer than 2 nodes with outgoing degree >= 2. This should be a quick check as we just need to pass the set of nodes in the component to `out_degree` and get value counts in the list. 

This leaves us with a subgraph containing only relevant connected components. Then, we create another subgraph, removing all division-involved edges. If we track the dividing nodes from the subgraph step above, this might help us avoid a second iteration. Once we have that subgraph, we should be able to just call `len` on the edges in each connected component to get all cell cycle lengths in the graph.

Reply:
I realized we need to keep strictly tracklets that start and end in nodes with outgoing degree > 1. I'm going to take a look at this on Thursday. I suspect there's a memory or quadratic issue somewhere in the traversal, so I'll profile and report back.